src/vivarium_uw_covid/components/seiir_beta.py

```diff
@@ -37,7 +37,79 @@ def beta_predict(coeffs, covs):
 
-def make_alternative_covariates(df_covs, loc_id, **alt_cov_values):
+# from https://github.com/ihmeuw/covid-model-seiir-pipeline/blob/master/src/covid_model_seiir_pipeline/core/utils.py#L128
+def beta_shift(beta_fit: pd.DataFrame,
+               beta_pred: np.ndarray,
+               window_size: Union[int, None] = None,
+               avg_over: Union[int, None] = None) -> Tuple[np.ndarray, float]:
+    """Calculate the beta shift.
+    Args:
+        beta_fit (pd.DataFrame): Data frame contains the date and beta fit.
+        beta_pred (np.ndarray): beta prediction.
+        windown_size (Union[int, None], optional):
+            Window size for the transition. If `None`, Hard shift no transition.
+            Default to None.
+        avg_over (Union[int, None], optional):
+            Final beta scale depends on the ratio between beta prediction over the
+            average beta over `avg_over` days. If `None`, final scale will be 1, means
+            return to the `beta_pred` completely. Default to None.
+    Returns:
+        Tuple[np.ndarray, float]: Predicted beta, after scaling (shift) and the initial scaling.
+    """
+    assert 'date' in beta_fit.columns, "'date' has to be in beta_fit data frame."
+    assert 'beta' in beta_fit.columns, "'beta' has to be in beta_fit data frame."
+    beta_fit = beta_fit.sort_values('date')
+    beta_fit = beta_fit['beta'].values
+
+    anchor_beta = beta_fit[-1]
+    scale_init = anchor_beta / beta_pred[0]
+
+    if avg_over is None:
+        scale_final = 1.0
+    else:
+        beta_history = beta_fit[-avg_over:]
+        scale_final = beta_history.mean() / beta_pred[0]
+
+    if window_size is not None:
+        assert isinstance(window_size, int) and window_size > 0, f"window_size={window_size} has to be a positive " \
+                                                                 f"integer."
+        scale = scale_init + (scale_final - scale_init)/window_size*np.arange(beta_pred.size)
+        scale[(window_size + 1):] = scale_final
+    else:
+        scale = scale_init
+
+    betas = beta_pred * scale
+
+    return betas, scale_init
+
+
+def beta_finalize(beta_pred, beta_fit):
+    """Calculate the final beta(t) values from beta_pred and beta_fit
+    
+    Parameters
+    ----------
+    beta_pred : pd.DataFrame with columns for covariates and rows for draws
+                e.g. the output of beta_predict
+    beta_fit : dict of pd.DataFrames with columns for beta and date,
+               e.g the output of load_beta_fit
+
+    Results
+    -------
+    returns pd.DataFrame of final beta values, with columns for each draw,
+    and rows for each day
+    """
+    beta_final = pd.DataFrame(columns=beta_pred.columns, index=beta_pred.index)
+    crossover_time = pd.Timestamp(beta_fit[0].date.max())
+    for draw in beta_fit.keys():
+        s_final, scale_init = beta_shift(beta_fit=beta_fit[draw],
+                                         beta_pred=beta_pred.loc[crossover_time:, draw],
```

Comment:
@collijk did I get this crossover_time right? I think it would be easy to mess it up.

Reply:
I'll have to look in detail at your data sources.  I'll take a look this evening.

Reply:
Your beta fit and beta pred should overlap on exactly one day (the initial condition).  So this should work out.